src/portable/nuvoton/nuc505/dcd_nuc505.c

```diff
@@ -579,15 +575,14 @@ void dcd_irq_handler(uint8_t rhport)
     else if (cep_state & USBD_CEPINTSTS_STSDONEIF_Msk)
     {
       /* given ACK from host has happened, we can now set the address (if not already done) */
-      if((USBD->FADDR != assigned_address) && (USBD->FADDR == 0)) USBD->FADDR = assigned_address;
-
-      if (configuration_changed)
+      if((USBD->FADDR != assigned_address) && (USBD->FADDR == 0))
```

Comment:
how about using the optional `dcd_edpt0_status_complete()` check for the complete of set_address and change it there. It is function called at the end of every Control transfer. Don't worry if it slows down the mcu, control transfer only happens in a very brief time of enumeration and rarely in the whole device activities.

https://github.com/hathach/tinyusb/blob/master/src/portable/microchip/samd/dcd_samd.c#L180